111-use-buildah-instead-of-kaniko.md

```diff
@@ -0,0 +1,179 @@
+    - `--cert-dir` - path to directory with certificates that should be used.
+    - `--creds` - the `[username[:password]]` to use to authenticate with the registry if required.
+    - `--decryption-key` - the `[key[:passphrase]]` to be used for decryption of images.
+    - `--env` - add env with value to the image.
+    - `--label` - add an image label to the image metadata.
+    - `--logfile` - log output which would be sent to standard output and standard error to the specified file instead of to standard output and standard error.
+    - `--manifest` - name of the manifest list to which the built image will be added.
+    - `--retry` - number of times to retry in case of failure during image pull (the base one).
+    - `--retry-delay` - duration between retry attempts.
+    - `--timestamp` - sets the "created" timestamp to the image configuration and manifest. When set, the "created" timestamp is always set to time specified, not generating new SHA.
+    - `--tls-verify` - skip TLS verification for insecure registries.
+- `additionalBuildahPushOptions` - for additional options to the `push` command. 
+  - Allowed options will be: 
+    - `--authfile` - path to the authentication file (usually created using the `buildah login`).
+    - `--cert-dir` - path to directory with certificates that should be used for connecting to registries.
+    - `--creds` - the `[username[:password]]` to use to authenticate with the registry if required.
+    - `--format` - control the format for the built image's manifest and configuration data. `oci` or `docker`.
+    - `--quiet` - when writing the output image, suppress progress output.
+    - `--remove-signatures` - don't copy signatures when pushing images.
+    - `--retry` - number of times to retry in case of failure during image pull (the base one).
+    - `--retry-delay` - duration between retry attempts.
+    - `--sign-by` - sign the pushed image using the GPG key that matches the specified fingerprint.
+    - `--tls-verify` - skip TLS verification for insecure registries.
+
+As for the Kaniko additional options, if the user-specified Buildah options will contain forbidden options (or not known), user will be notified by message inside the `.status` section of `KafkaConnect` resource, the `InvalidResourceException` will be thrown (logged inside the operator log) and the build will fail.
+
+If Buildah is enabled through the feature gate and the user provides Kaniko-specific options, those options will be ignored, and a warning will be reported in the .status section of the KafkaConnect CR.
+Once the feature gate will be moved to beta, the additional Kaniko options field will be deprecated.
+
+Finally, this feature will be available on Kubernetes only - it will not be available on OpenShift, which is mentioned as one of the rejected alternative.
+So the implementation and usage of OpenShift Build API will be kept.
+
```

Comment:
Regarding the above paragraph could we have a YAML example?

Reply:
YAML example of what exactly? Sorry I have some YAMLs that I used for the PoC on OCP - ClusterRoles, Rolebindings... so I'm not sure what exactly do you mean :D